.github/actions/setup_environment/action.yml

```diff
@@ -32,43 +38,59 @@ runs:
+        # - manually triggered by means of a custom cache version token
+        # - on a weekly basis
+        # - any of the integration requirements change (a hash of the
+        # __init__.py files is included in the cache key)
+        custom_cache_key_element: ${{ inputs.cache_version }}-${{steps.date.outputs.week}}-${{inputs.install_integrations}}-${{ hashFiles('src/zenml/integrations/*/__init__.py') }}
 
-    - name: Load cached venv
-      id: cached-virtualenv
-      uses: actions/cache@v2.1.6
+    - uses: syphar/restore-pip-download-cache@v1
       with:
-        path: .venv
-        # Cache the complete venv dir for a given os, python version,
-        # pyproject.toml, the current calendar week and the cache version.
-        # Providing a different value for the cache version enables manual
-        # invalidation of the cache in case it is in some erroneous state.
-        key: venv-${{ runner.os }}-python-${{ inputs.python-version }}-${{ hashFiles('pyproject.toml') }}-${{ steps.date.outputs.date }}-${{ hashFiles('cache_version') }}
+        requirement_files: 'pyproject.toml'
+        # The pip download cache can be updated on a weekly basis as new packages
+        # don't appear that often
+        custom_cache_key_element: ${{ inputs.cache_version }}-${{steps.date.outputs.week}}
+
+    - name: Install ZenML
+      if: steps.cache-virtualenv.outputs.cache-hit != 'true'
+      shell: bash
+      run: |
+        python -m pip install --upgrade pip
+        pip install -e .[server,templates,secrets-aws,secrets-gcp,secrets-azure,secrets-hashicorp,dev]
 
-    - name: Set virtualenv activation script env variable (windows)
+    - name: Install basic ZenML integrations
+      if: steps.cache-virtualenv.outputs.cache-hit != 'true' && inputs.install_integrations == 'yes'
       shell: bash
-      run: echo "VENV=.venv/scripts/activate" >> "$GITHUB_ENV"
-      if: runner.os == 'Windows'
+      run: |
+        zenml integration export-requirements --output-file integration-requirements.txt --ignore-integration feast --ignore-integration label_studio --ignore-integration bentoml --ignore-integration seldon --ignore-integration kserve
+        pip install -r integration-requirements.txt
+        rm integration-requirements.txt
 
-    - name: Set virtualenv activation script env variable (unix)
+      # Refresh the ZenML installation if using a cached virtualenv or after
+      # installing integrations
+    - name: Refresh ZenML installation
+      if: steps.cache-virtualenv.outputs.cache-hit == 'true' || inputs.install_integrations == 'yes'
       shell: bash
-      run: echo "VENV=.venv/bin/activate" >> "$GITHUB_ENV"
-      if: runner.os != 'Windows'
+      run: |
+        pip install -e .[server,templates,secrets-aws,secrets-gcp,secrets-azure,secrets-hashicorp,dev]
+
+        # install airflow requirements
+        # pip install --upgrade "numpy>=1.21.0,<1.24.0"
+        # pip install apache-airflow-providers-docker
```

Comment:
What are these here for?

Reply:
Left overs from the old CI, already cleaned up.